File: stock-market-service/src/main/java/rs/edu/raf/scheduler/OrderProcessScheduler.java
package rs.edu.raf.scheduler;

import lombok.AllArgsConstructor;
import org.springframework.scheduling.annotation.Scheduled;
import org.springframework.stereotype.Component;
import rs.edu.raf.service.OrderService;

@Component
@AllArgsConstructor
public class OrderProcessScheduler {
    private OrderService orderService;

    @Scheduled(fixedDelay = 60000)
    public void processOrders() {
        orderService.processOrders();
    }
}

File: stock-market-service-kotlin/src/main/kotlin/org/example/stockmarketservicekotlin/scheduler/OrderProcessScheduler.kt
package org.example.stockmarketservicekotlin.scheduler

import org.example.stockmarketservicekotlin.service.OrderService
import org.springframework.scheduling.annotation.Scheduled
import org.springframework.stereotype.Component

@Component
class OrderProcessScheduler(private val orderService: OrderService) {
    @Scheduled(fixedDelay = 60000)
    fun processOrders() {
        orderService.processOrders()
    }
}